go/exorelay/main.go

```diff
@@ -57,3 +66,42 @@ func (exoRelay *ExoRelay) Send(eventName string, payload map[string]interface{})
 
 	return websocket.Message.Send(exoRelay.socket, serializedBytes)
 }
+
+func (exoRelay *ExoRelay) receiveMessage(message structs.Message) {
+	if exoRelay.messageHandlerMapping == nil {
+		return
+	}
+	if exoRelay.messageHandlerMapping[message.Name] == nil {
+		return
+	}
+	exoRelay.messageHandlerMapping[message.Name](message.Payload)
+}
+
+func (exoRelay *ExoRelay) listenForMessages() {
+	for {
+		message, err := exoRelay.readMessage()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			fmt.Println("Error reading message from websocket:", err)
+			break
+		} else {
+			exoRelay.receiveMessage(message)
+		}
+	}
+}
+
+func (exoRelay *ExoRelay) readMessage() (structs.Message, error) {
+	var bytes []byte
+	if err := websocket.Message.Receive(exoRelay.socket, &bytes); err != nil {
+		return structs.Message{}, err
+	}
+
+	var unmarshaled structs.Message
+	err := json.Unmarshal(bytes, &unmarshaled)
+	if err != nil {
+		return structs.Message{}, err
+	}
+
+	return unmarshaled, nil
+}
```

Comment:
This might be because the `websocket` in ExocomMock is a different version than the one used here. And why it is recommended that libraries don't vendor their dependencies, but use the one from the binary that uses them. https://github.com/Originate/guide_private/pull/61 proposes this rule, based on literature in the Go community. Does deleting the vendor folder in ExocomMock solve this issue?